lfs.c

```diff
@@ -3064,11 +3064,26 @@ static lfs_soff_t lfs_file_rawseek(lfs_t *lfs, lfs_file_t *file,
     // find new pos
     lfs_off_t npos = file->pos;
     if (whence == LFS_SEEK_SET) {
```

Comment:
LFS_SEEK_SET is a bit of a special case, and should probably be documented better.

One problem with using a signed (`lfs_soff_t`) integer to represent file offsets is that this doesn't capture the full range of files stored with an unsigned file size.

Unsigned file sizes are NOT the default (`LFS_FILE_MAX` defaults to `2147483647`), and if enabled (`LFS_FILE_MAX = 4294967294`) it causes problems, but it's still possible to use with a limited set of littlefs APIs.

To help this, `lfs_file_seek` with `LFS_SEEK_SET` treats its offset as unsigned, even though it is a passed as a signed integer. So `off < 0` in this case isn't an error, it should set the file to an extremely large value.

This does risk some confusing cases if `lfs_file_seek` is called with a negative value, but most `LFS_SEEK_SET` operations are usually calculated using absolute, unsigned values anyways.